diff -ruN postgresql-18.1-orig/src/include/c.h postgresql-18.1/src/include/c.h
--- postgresql-18.1-orig/src/include/c.h	2025-11-28 18:54:52.627048100 +0100
+++ postgresql-18.1/src/include/c.h	2025-11-28 18:56:22.269355100 +0100
@@ -59,7 +59,6 @@
 #include "pg_config_os.h"		/* config from include/port/PORTNAME.h */
 
 /* System header files that should be available everywhere in Postgres */
-#include <inttypes.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
