diff -urEb mariadb-connector-c-3.1.11-src.orig/include/ma_global.h mariadb-connector-c-3.1.11-src/include/ma_global.h
--- mariadb-connector-c-3.1.11-src.orig/include/ma_global.h	2020-12-16 13:47:31.950009259 +0100
+++ mariadb-connector-c-3.1.11-src/include/ma_global.h	2020-12-16 13:50:23.705910282 +0100
@@ -606,7 +606,9 @@
 #error "Neither int or long is of 4 bytes width"
 #endif
 
-#if !defined(HAVE_ULONG) && !defined(HAVE_LINUXTHREADS) && !defined(__USE_MISC)
+#if defined(__LSB_VERSION__)
+typedef unsigned long	ulong;	/* Short for unsigned long */
+#elif !defined(HAVE_ULONG) && !defined(HAVE_LINUXTHREADS) && !defined(__USE_MISC)
 typedef unsigned long	ulong;	/* Short for unsigned long */
 #endif
 #ifndef longlong_defined
diff -urEb mariadb-connector-c-3.1.11-src.orig/include/mysql/client_plugin.h mariadb-connector-c-3.1.11-src/include/mysql/client_plugin.h
--- mariadb-connector-c-3.1.11-src.orig/include/mysql/client_plugin.h	2020-12-16 13:47:31.950009259 +0100
+++ mariadb-connector-c-3.1.11-src/include/mysql/client_plugin.h	2020-12-16 13:50:37.913900837 +0100
@@ -180,7 +180,7 @@
   @retval
   a pointer to the loaded plugin, or NULL in case of a failure
 */
-struct st_mysql_client_plugin *
+struct st_mysql_client_plugin * STDCALL
 mysql_load_plugin(struct st_mysql *mysql, const char *name, int type,
                   int argc, ...);
 
diff -urEb mariadb-connector-c-3.1.11-src.orig/include/mysql.h mariadb-connector-c-3.1.11-src/include/mysql.h
--- mariadb-connector-c-3.1.11-src.orig/include/mysql.h	2020-12-16 13:47:31.950009259 +0100
+++ mariadb-connector-c-3.1.11-src/include/mysql.h	2020-12-16 13:54:01.145749406 +0100
@@ -467,7 +467,7 @@
   MYSQL_CLIENT_PLUGIN_HEADER
 };
 
-struct st_mysql_client_plugin *
+struct st_mysql_client_plugin * STDCALL
 mysql_load_plugin(struct st_mysql *mysql, const char *name, int type,
                   int argc, ...);
 struct st_mysql_client_plugin * STDCALL
Only in mariadb-connector-c-3.1.11-src/include: mysql.h~
diff -urEb mariadb-connector-c-3.1.11-src.orig/plugins/pvio/pvio_socket.c mariadb-connector-c-3.1.11-src/plugins/pvio/pvio_socket.c
--- mariadb-connector-c-3.1.11-src.orig/plugins/pvio/pvio_socket.c	2020-12-16 13:47:31.954009258 +0100
+++ mariadb-connector-c-3.1.11-src/plugins/pvio/pvio_socket.c	2020-12-16 13:50:23.705910282 +0100
@@ -60,6 +60,11 @@
 #define IS_SOCKET_EINTR(err) 0
 #endif
 
+#if defined(__LSB_VERSION__)
+// WARNING: This definition might break true Linux Standard Base compatibility!
+#  define MSG_DONTWAIT 0x40 /* Nonblocking IO.  */
+#endif
+
 #ifndef SOCKET_ERROR
 #define SOCKET_ERROR -1
 #endif
